supporting-blog-content/customer-success-example/chat-example/app/api/chat/route.ts

```diff
@@ -0,0 +1,53 @@
+import { openai } from '@ai-sdk/openai';
+import { convertToCoreMessages, streamText, tool } from 'ai';
+const { Client } = require('@elastic/elasticsearch-serverless')
+import { z } from 'zod';
+
+const client = new Client({
+  node: process.env.ELASTICSEARCH_URL, // serverless project URL
+  auth: { apiKey: process.env.ES_API_KEY }, // project API key
+})
+
+// add a function findRelevantContent to retrieve relevant content from the knowledge base
+async function findRelevantContent(question: string) {
+  const body = await client.search({
+    size: 3,
+    index: 'search-faq',
+    body: {
+      query: {
+        "semantic": {
+            "field": "semantic",
+            "query": question
+          }
+      }
+    }
+  });
+  return body.hits.hits.map((hit: any) => ({
+    content: hit._source.content
+  }));
+}
+
+export async function POST(req: Request) {
+
```

Comment:
Sorry for being nitpicky, but consistent spacing would be useful even though it's a demo application.
```suggestion
```